From 2ed855954f54f622b180ce007c044eb6c6119872 Mon Sep 17 00:00:00 2001
From: Yanjiang Jin <yanjiang.jin@windriver.com>
Date: Thu, 14 Aug 2014 13:42:21 +0800
Subject: [PATCH 7/8] bcm-xlp: NAE: enable the functions for set MTU and MAC
 ADDR

Enable the functions for set MTU and MAC address.

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/netlogic/nae/xlpge_nae.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/drivers/netlogic/nae/xlpge_nae.c b/drivers/netlogic/nae/xlpge_nae.c
index e9a5bb8..a2275a0 100644
--- a/drivers/netlogic/nae/xlpge_nae.c
+++ b/drivers/netlogic/nae/xlpge_nae.c
@@ -94,8 +94,8 @@ module_param(force_tso, int, 0);
 static uint32_t lnx_frfifo_normal_mask[NLM_MAX_NODES][MAX_NAE_PERNODE];
 static uint32_t lnx_frfifo_jumbo_mask[NLM_MAX_NODES][MAX_NAE_PERNODE];
 
-int enable_jumbo = 0;
-module_param(enable_jumbo, int, 0);
+int enable_jumbo = 1;
+module_param(enable_jumbo, int, 1);
 struct p2p_desc_mem p2p_desc_mem[NR_CPUS] __cacheline_aligned;
 static unsigned int phys_cpu_map[NLM_MAX_NODES];
 struct timer_list phy_int_timer;
@@ -1321,7 +1321,7 @@ static void  nlm_xlp_nae_tx_timeout (struct net_device *dev)
 	nae_print(NAE_DBG_ERROR, "%s: Transmit timed out\n", dev->name);
 	return;
 }
-#if 0
+
 static int nlm_xlp_nae_set_hwaddr(struct net_device *dev, void *p)
 {
 	struct sockaddr *addr = (struct sockaddr *)p;
@@ -1349,7 +1349,6 @@ static int nlm_xlp_nae_set_hwaddr(struct net_device *dev, void *p)
 
 	return rc;
 }
-#endif
 
 #ifdef ENABLE_NAE_PIC_INT
 /**********************************************************************
@@ -1392,7 +1391,7 @@ static const struct net_device_ops nlm_xlp_nae_ops = {
 	.ndo_do_ioctl			= nlm_xlp_nae_ioctl,
 	.ndo_tx_timeout 		= nlm_xlp_nae_tx_timeout,
 	.ndo_change_mtu			= nlm_xlp_nae_change_mtu,
-	//.ndo_set_mac_address		= nlm_xlp_nae_set_hwaddr,
+	.ndo_set_mac_address		= nlm_xlp_nae_set_hwaddr,
 	.ndo_get_stats 			= nlm_xlp_mac_get_stats,
 	.ndo_select_queue		= nlm_select_queue,
 };
-- 
1.9.1

